cityofaustin/vision-zero
# Add more layers to fancy helper script

## Associated issues

- https://github.com/cityofaustin/atd-data-tech/issues/18735
- https://github.com/cityofaustin/atd-data-tech/issues/18736
- https://github.com/cityofaustin/atd-data-tech/issues/19407
- https://github.com/cityofaustin/atd-data-tech/issues/19408

## Summary of changes

- The script now truncates and inserts all rows for all tables
- Remove old council districts and engineering area toolbox scripts
- Rename `updated_at` to `created_at` on council district and engineering areas layers and use `timestamptz`
- Make engineering areas column names match the AGOL layer
- Update the helper script to deal with council districts having two separate features for district 10 🥴
- Add a new `--save` flag for debugging the helper script
- Add zip codes and APD sectors tables to the DB and add them to the helper script

## Out of scope

- New triggers to set crash properties based on any of the new layers (future issue)
- Moving layers into a separate DB schema (future issue)

## Testing

**URL to test:** Local

1. Start your local stack and apply migrations + metadata

2. Save your env file as `local.env` as described in the README.

3. Activate your `node` version and test each of the layer commands:

```
nvm use
```

```
node --env-file=local.env src/load_agol_layer.js -l apd_sectors
```

```
node --env-file=local.env src/load_agol_layer.js -l council_districts
```

```
node --env-file=local.env src/load_agol_layer.js -l engineering_areas
```

```
node --env-file=local.env src/load_agol_layer.js -l zip_codes
```

```
node --env-file=local.env src/load_agol_layer.js -l non_coa_roadways
```

4. Start the VZE, and edit a crash by moving it's location outside the Austin city limits, and then edit it again to save it's location somewhere within the city limits.

5. Inspect the change log at the bottom of the crash details page to verify that the council_district and engineering area were updated correctly based on your edits.



---

#### Ship list
